doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        static CatalogPackage TryCreateFromManifest(String manifestPath);
+
+        /// IMPLEMENTATION NOTE: PackageProperty fields from AppInstallerRepositorySearch.h
+        /// Gets a property of this package.
+        String Id { get; };
+        String Name { get; };
+
+        /// Gets the installed package information if the package is installed.
+        PackageVersionInfo InstalledVersion{ get; };
+
+        /// Gets all available versions of this package.
+        /// The versions will be returned in sorted, descending order.
+        ///  Ex. { 4, 3, 2, 1 }
+        Windows.Foundation.Collections.IVectorView<PackageVersionId> AvailableVersions { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersionInfo LatestAvailableVersion { get; };
+
+        /// Gets a specific version of this package.
+        PackageVersionInfo GetAvailableVersion(PackageVersionId versionKey);
+
+        /// Gets a value indicating whether an available version is newer than the installed version.
+        Boolean IsUpdateAvailable { get; };
+
+        /// Gets whether the package is installing
+        Boolean IsInstalling { get; };
+
+        /// DESIGN NOTE:
+        /// IsSame from IPackage in AppInstallerRepositorySearch is not implemented in V1.
+        /// Determines if the given IPackage refers to the same package as this one.
+        /// virtual bool IsSame(const IPackage*) const = 0;
+    }
+
+    /// IMPLEMENTATION NOTE: CompositeSearchBehavior from AppInstallerRepositorySource.h
+    /// Search behavior for composite sources.
+    /// Only relevant for composite sources with a local source, not for aggregates of multiple available sources.
+    /// Installed and available packages in the result are always correlated when possible.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum CompositeSearchBehavior
+    {
+        /// Search only installed packages
+        InstalledPackages,
+        /// Search only installing packages
+        InstallingPackages,
+        /// Search only local packages (installed and installing)
+        AllLocalPackages,
+        /// Search both local and remote sources.
+        AllPackages,
+    };
+
+    /// IMPLEMENTATION NOTE: MatchType from AppInstallerRepositorySearch.h
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum MatchType
+    {
+        Exact,
+        CaseInsensitive,
+        StartsWith,
+        Fuzzy,
+        Substring,
+        FuzzySubstring,
+        Wildcard,
+    };
+    /// IMPLEMENTATION NOTE: PackageMatchField from AppInstallerRepositorySearch.h
+    /// The field to match on.
+    /// The values must be declared in order of preference in search results.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    enum PackageMatchField
+    {
+        AppCatalogDefined,
+        Id,
+        Name,
+        Moniker,

```

Comment:
What are Moniker, Command, Tag?

Reply:
Various fields in the yaml designed to make the apps searchable https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv0.1.md I'm not aware of any guidance to help developers decide what to throw in each of them, as in say, a list of popular tags.